### PR TITLE
Bloomfiltertrie

### PR DESCRIPTION
**Pull request by Colelyman**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

As per the guidelines, I would like to request a review from @bioconda/core because this is my first recipe. Thanks!

### CODE REVIEW

**Comment by druvus** [comment]
@Colelyman An alternative here would be enabling openmp support again and use gcc for both linux and osx as clang doesn't support it.

**Comment by mbargull** [comment]
Would depending on
```yaml
- llvm
- llvm-openmp
```
help? I haven't had the change to try that one yet.

File: recipes/bloomfiltertrie/popcnt.patch [hunk 1]
```diff
@@ -0,0 +1,22 @@
+diff --git a/src/intersection.c b/src/intersection.c
+index 56599bb..443418e 100755
+--- a/src/intersection.c
++++ b/src/intersection.c
+@@ -137,7 +137,7 @@ uint32_t* intersection_uint32_SIMD(uint32_t* list_a, uint32_t* list_b) {
+         //[ copy out common elements
+         __m128i p = _mm_shuffle_epi8(v_a, shuffle_mask[mask]);
+         _mm_storeu_si128((__m128i*)&C[count], p);
+-        count += _mm_popcnt_u32(mask);
++        count += __builtin_popcountl(mask);
```
Comment (mbargull) [inline recipes/bloomfiltertrie/popcnt.patch]: Is this a behavioral change? The original code does a 32-bit pop-count, your change would make that 64-bit.

**Comment by Colelyman** [comment]
Unfortunately depending on `llvm-openmp # [osx[` didn't work. @druvus would removing the `llvm` dependency completely have the MacOS use `gcc` or is there something different that I should do?

**Comment by Colelyman** [comment]
@druvus eliminating the `llvm` dependency worked! Thank you

File: recipes/bloomfiltertrie/popcnt.patch (hunk 1 above)
Comment (mbargull) [reply recipes/bloomfiltertrie/popcnt.patch]: Due to compatibility reasons, I don't think we can use SSE 4.2 yet; though, TBH, I'm not 100 % sure what the expected minimum capabilities/extensions we target are.
Here, I think you can just use `__builtin_popcount` (minus the `l`ong); that's also what GCC would do anyway if SSE >=4.1 and POPCNT were enabled: https://github.com/gcc-mirror/gcc/blob/gcc-4_8_5-release/gcc/config/i386/popcntintrin.h#L35.

**Comment by mbargull** [comment]
> Unfortunately depending on `llvm-openmp # [osx]` didn't work.

Was worth a shot -- thanks for trying it out. Using `gcc` should be fine.

**Comment by Colelyman** [comment]
@mbargull anything else that I should do before I squash and merge?

**Comment by mbargull** [comment]
Looks good, thank you @Colelyman!

**Pull request merged**